File: CPP/Fixed_Heap_Dynamic.cpp
#include <iostream>
using namespace std;

int main() {
    int size;
    cout << "Enter array size: ";
    cin >> size;

    int* heapArray = new int[size];  // Allocated at runtime
    for (int i = 0; i < size; i++)
        heapArray[i] = 0;

    heapArray[1] = 25;
    heapArray[3] = 50;

    cout << "Fixed Heap Dynamic Array: ";
    for (int i = 0; i < size; i++)
        cout << heapArray[i] << " ";
    cout << endl;

    delete[] heapArray;  // Free heap memory
    return 0;
}

File: Python/Fixed_Heap_Dynamic.py

class FixedHeapArray:
    def __init__(self, size):
        self.size = size
        self.array = [0] * size  # Allocate at runtime with fixed size
        print(f"Fixed Heap Dynamic Array of size {size} created.")

    def update(self, index, value):
        if 0 <= index < self.size:
            self.array[index] = value
            print(f"Updated index {index} to {value}: {self.array}")
        else:
            print("Index out of bounds.")

if __name__ == "__main__":
    heap_array = FixedHeapArray(4)
    heap_array.update(1, 20)
    heap_array.update(3, 40)
